homeassistant/components/isy994/button.py

```diff
@@ -36,47 +37,62 @@ async def async_setup_entry(
     async_add_entities(entities)
 
 
-class ISYNodeQueryButtonEntity(ButtonEntity):
-    """Representation of a device query button entity."""
+class ISYNodeButtonEntity(ButtonEntity):
+    """Additional methods for ISYNodeButton classes."""
 
     _attr_should_poll = False
-    _attr_entity_category = EntityCategory.CONFIG
     _attr_has_entity_name = True
 
     def __init__(self, node: Node | ISY, base_unique_id: str) -> None:
-        """Initialize a query ISY device button entity."""
+        """Initialize a ISY device button entity."""
         self._node = node
+        self._attr_entity_registry_enabled_default = getattr(node, "enabled", True)
+        self._attr_device_info = DeviceInfo(
+            identifiers={(ISY994_DOMAIN, base_unique_id)}
+        )
+
+    def check_disabled(self) -> bool:
+        """Warn the user if the device is disabled in the ISY."""
+        if not getattr(self._node, "enabled", True):
+            raise HomeAssistantError(
+                f"Attempted to send command to device disabled in ISY. Use isy994.send_node_command to enable {self.entity_id}"
+            )
```

Comment:
Would it make sense to have an `EntityCategory.CONFIG` `switch` entity to enable/disable the node instead?

Reply:
Not opposed to adding a switch for this to each parent device. With the new button logic we already identify the parent devices.

What's the best way to handle the other entities when the node is disabled in the ISY? Still leave the errors?

I'm still kind of tossing around just not adding the entities if its disabled--that would leave existing entities in "Unavailable" state. I guess we could unload just the entities for that device? Although the integrations not set up for that right now.

FYI - my personal use case for these is all of my temporary plugs for x-mas, they get disabled in the ISY when unplugged so it doesn't nag about the errors all year, but you don't lose all the entities in Home Assistant either.

Reply:
Also: I don't do anything with sensor and binary_sensor nodes that are disabled, but should they all be updated to return state unknown if disabled?

Reply:
Another thing that could be done is add them as `disabled_by` `RegistryEntryDisabler.INTEGRATION` and when they get enabled in the UI enable them on the ISY

I'm not 100% sure that they could be enabled in the UI though as its been a while since I played with that

Reply:
Backed up to draft for now. I did some work today on the PyISY side https://github.com/automicus/PyISY/releases/tag/v3.1.1 which will allow for receiving the nodes being enabled/disabled without polling. This is also the first steps to being able to update Home Assistant entities live when changed on the ISY without having to reload the module.

I like the idea of using a disable/enable config switch. I will have to play around to see if there's an easy way to disable/enable the rest of the entities for a device if the device is disabled (and leave the enable toggle). Z-Wave JS does this when a device is marked dead (everything but the Ping goes unavailable).

I'll come back to this when I get time. If you get 2-sec please take a look at the [changes to logging](https://github.com/automicus/PyISY/pull/353) on the PyISY and make sure I didn't do anything that's going to have negative upstream effects -- the module logging should only get configured if run as a module, everything else gets a `NullHandler` if there's no upstream logger.